src/Microsoft.AspNet.OData.Shared/EnableQueryAttribute.cs

```diff
@@ -0,0 +1,783 @@
+        public bool EnsureStableOrdering
+        {
+            get
+            {
+                return _querySettings.EnsureStableOrdering;
+            }
+            set
+            {
+                _querySettings.EnsureStableOrdering = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating how null propagation should
+        /// be handled during query composition.
+        /// </summary>
+        /// <value>
+        /// The default is <see cref="HandleNullPropagationOption.Default"/>.
+        /// </value>
+        public HandleNullPropagationOption HandleNullPropagation
+        {
+            get
+            {
+                return _querySettings.HandleNullPropagation;
+            }
+            set
+            {
+                _querySettings.HandleNullPropagation = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether constants should be parameterized. Parameterizing constants
+        /// would result in better performance with Entity framework.
+        /// </summary>
+        /// <value>The default value is <c>true</c>.</value>
+        public bool EnableConstantParameterization
+        {
+            get
+            {
+                return _querySettings.EnableConstantParameterization;
+            }
+            set
+            {
+                _querySettings.EnableConstantParameterization = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum depth of the Any or All elements nested inside the query. This limit helps prevent
+        /// Denial of Service attacks.
+        /// </summary>
+        /// <value>
+        /// The maximum depth of the Any or All elements nested inside the query. The default value is 1.
+        /// </value>
+        public int MaxAnyAllExpressionDepth
+        {
+            get
+            {
+                return _validationSettings.MaxAnyAllExpressionDepth;
+            }
+            set
+            {
+                _validationSettings.MaxAnyAllExpressionDepth = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum number of nodes inside the $filter syntax tree.
+        /// </summary>
+        /// <value>The default value is 100.</value>
+        public int MaxNodeCount
+        {
+            get
+            {
+                return _validationSettings.MaxNodeCount;
+            }
+            set
+            {
+                _validationSettings.MaxNodeCount = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum number of query results to send back to clients.
+        /// </summary>
+        /// <value>
+        /// The maximum number of query results to send back to clients.
+        /// </value>
+        public int PageSize
+        {
+            get
+            {
+                return _querySettings.PageSize ?? default(int);
```

Comment:
What's the behavior when the max page size is default(int), which is 0? Why is PageSize the only property with the ?? operator?